src/utils/keyHelpers/acl.ts

```diff
@@ -0,0 +1,78 @@
+import * as debug from '../../debug'
+import { store } from 'solid-logic'
+
+/**
+ * set ACL
+ * @param keyDoc
+ * @param aclBody
+ */
+export async function setAcl (keyDoc: string, aclBody: string) {
+  // Some servers don't present a Link http response header
+  // if the container doesn't exist yet, refetch the resource
+
+  await store.fetcher.load(keyDoc)
+
+  // FIXME: check the Why value on this quad:
+  debug.log(store.statementsMatching(store.sym(keyDoc), store.sym('http://www.iana.org/assignments/link-relations/acl')))
+  const keyAclDoc = store.any(store.sym(keyDoc), store.sym('http://www.iana.org/assignments/link-relations/acl'))
+  if (!keyAclDoc) {
+    throw new Error('Key ACL doc not found!')
+  }
+
+  // delete READ only keyAclDoc. This is possible if the webId is an owner
+  try {
+    const response = await store.fetcher.webOperation('DELETE', keyAclDoc.value) // this may fail if webId is not an owner
```

Comment:
@bourgeoa I wanted to find out more about why we do this. It looks like we try to delete the acl, but then "PUT" it again with the read turtle below. Can you tell me more about it?

Reply:
This is because the ACL exists and is Read only.
We cannot Write the new key with such ACL